### PR TITLE
[SYCL] Enable use of variadic templates in SYCL kernel names.

### PR DESCRIPTION
**Pull request by alexeyvoronov**
Added support for template parameters pack processing to the
clang::TypeName::getFullyQualifiedTemplateArgument method.

Signed-off-by: Alexey Voronov <alexey.voronov@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@agozillon, does this patch covers #46?

File: clang/lib/AST/QualTypeNames.cpp [hunk 1]
```diff
@@ -109,6 +109,10 @@ static bool getFullyQualifiedTemplateArgument(const ASTContext &Ctx,
       Arg = TemplateArgument(QTFQ);
       Changed = true;
     }
+  } else if (Arg.getKind() == TemplateArgument::Pack) {
+    for (const auto &It : Arg.pack_elements()){
+      Changed |= getFullyQualifiedTemplateArgument(Ctx, const_cast<TemplateArgument&>(It), WithGlobalNsPrefix);
```
Comment (AlexeySachkov) [inline clang/lib/AST/QualTypeNames.cpp]: Please wrap long lines

Comment (alexeyvoronov) [reply clang/lib/AST/QualTypeNames.cpp]: fixed.

**Comment by alexeyvoronov** [comment]
> @agozillon, does this patch covers #46?

Test cases from that patch are successfully passed with this patch.

**Comment by alexeyvoronov** [comment]
Those tests are outdated, I updated it. They're here. If it is necessary I can add it.
[test_cases_46.txt](https://github.com/intel/llvm/files/3103562/test_cases_46.txt)

**Comment by bader** [comment]
> Those tests are outdated, I updated it. They're here. If it is necessary I can add it.
> [test_cases_46.txt](https://github.com/intel/llvm/files/3103562/test_cases_46.txt)

Do they cover more code than test case in your patch?

**Comment by alexeyvoronov** [comment]
I guess, yes.

**Comment by agozillon** [comment]
LGTM, better fix than my own, even adds to upstream Clang, I have no problems using this over my own pull request.

If you add the tests I added that'd be nice, but if you don't then that's not a problem either.

**Comment by alexeyvoronov** [comment]
> LGTM, better fix than my own, even adds to upstream Clang, I have no problems using this over my own pull request.
> 
> If you add the tests I added that'd be nice, but if you don't then that's not a problem either.

Thank you. I will apply your test cases with a little changes.

**Pull request merged**